skunkworks_crow/src/main/java/org/odk/share/bluetooth/BluetoothReceiver.java

```diff
@@ -33,25 +33,27 @@ public void onReceive(Context context, Intent intent) {
             return;
         }
 
-        BluetoothDevice bluetoothDevice = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
-        if (bluetoothDevice != null) {
-            Timber.d("BluetoothDevice: %s, Address: %s", bluetoothDevice.getName(), bluetoothDevice.getAddress());
-            switch (action) {
-                case BluetoothDevice.ACTION_FOUND:
-                    short rssi = intent.getShortExtra(BluetoothDevice.EXTRA_RSSI, Short.MAX_VALUE);
-                    Timber.d("EXTRA_RSSI: %s", rssi);
-                    bluetoothReceiverListener.onDeviceFound(bluetoothDevice);
-                    break;
-                case BluetoothAdapter.ACTION_DISCOVERY_STARTED:
-                    Timber.d("bluetooth devices discovery started...");
-                    bluetoothReceiverListener.onDiscoveryStarted();
-                    break;
-                case BluetoothAdapter.ACTION_DISCOVERY_FINISHED:
-                    Timber.d("bluetooth devices discovery finished...");
-                    bluetoothReceiverListener.onDiscoveryFinished();
-                    break;
+        // if the discovery started, update the ui in activity.
+        if (BluetoothAdapter.ACTION_DISCOVERY_STARTED.equals(action)) {
```

Comment:
I think `switch `case would be better in this case, it will make it easy to understand.

Reply:
I have an issue using `swith-case`, the `ACTION_DISCOVERY_STARTED` cannot be received, so I changed them to `if-else`, I think that is not a big deal.

Reply:
That's odd, maybe related to the `break`.

Reply:
That's strange, okay I'll take a look, anyways if you've added `if` statements then it would be better if you can make these individual stmts as `if else if..`, right now they all are written as individual `if` statements.

Reply:
Thanks, no hurry, let me have a try with `switch`.